### PR TITLE
watchman 2021.05.31.00

### PR DESCRIPTION
**Pull request by carlocab**
Created with `brew bump-formula-pr`.

### CODE REVIEW

File: Formula/watchman.rb [hunk 1]
```diff
@@ -1,28 +1,11 @@
 class Watchman < Formula
   desc "Watch files and take action when they change"
   homepage "https://github.com/facebook/watchman"
+  url "https://github.com/facebook/watchman/archive/v2021.05.31.00.tar.gz"
```
Comment (carlocab) [inline Formula/watchman.rb:4]: They seem to have switched their version scheme to using dates about a year ago.

File: Formula/watchman.rb [hunk 2]
```diff
@@ -1,28 +1,11 @@
...
-      sha256 "73990f0c7bd434d04fd5f1310b97c5f8599793007bd31ae438c2ba0211fb2c43"
-    end
-  end
-
-  # The Git repo contains a few tags like `2020.05.18.00`, so we have to
-  # restrict matching to versions with two to three parts (e.g., 1.2, 1.2.3).
-  livecheck do
-    url :stable
-    regex(/^v?(\d+(?:\.\d+){,2})$/i)
-  end
```
Comment (carlocab) [inline Formula/watchman.rb:24]: We no longer need to restrict the tags matched, so I think this is no longer needed.

**Comment by carlocab** [comment]
Ok, this is fetching and building its own Boost. Not cool.

File: Formula/watchman.rb [hunk 3]
```diff
@@ -31,34 +14,30 @@ class Watchman < Formula
     sha256 high_sierra:   "e872c3aae64c3b78197de9f12e272bebd5d20c316a120916f59a5f1cd2fac039"
   end
 
-  depends_on "autoconf" => :build
-  depends_on "automake" => :build
-  depends_on "libtool" => :build
+  depends_on "cmake" => :build
   depends_on "pkg-config" => :build
+  depends_on "rust" => :build
```
Comment (carlocab) [inline Formula/watchman.rb:19]:
```
[  3%] Building Rust executable 'watchmanctl'...
cd /tmp/watchman-20210602-41855-joxfes/watchman-2021.05.31.00/cli && /opt/homebrew/Cellar/cmake/3.20.3/bin/cmake -E remove -f /tmp/watchman-20210602-41855-joxfes/watchman-2021.05.31.00/cli/Cargo.lock
cd /tmp/watchman-20210602-41855-joxfes/watchman-2021.05.31.00/cli && /opt/homebrew/Cellar/cmake/3.20.3/bin/cmake -E env CARGO_TARGET_DIR=/tmp/watchman-20210602-41855-joxfes/watchman-2021.05.31.00/build/cli cargo build --release -p watchmanctl
No such file or directory
make[2]: *** [cli/CMakeFiles/watchmanctl.cargo] Error 1
make[1]: *** [cli/CMakeFiles/watchmanctl.cargo.dir/all] Error 2
```

File: Formula/watchman.rb [hunk 4]
```diff
@@ -78,6 +57,6 @@ def post_install
   end
 
   test do
-    assert_equal(/(\d+\.\d+\.\d+)/.match(version)[0], shell_output("#{bin}/watchman -v").chomp)
+    assert_equal(version.to_s, shell_output("#{bin}/watchman -v").chomp)
```
Comment (carlocab) [inline Formula/watchman.rb:60]: Tried to turn this into a real test. Unfortunately it doesn't work with the sandbox:
```
2021-06-03T05:38:10,359: [cli] Failed to open /Users/carlocab/Library/LaunchAgents/com.github.facebook.watchman.plist for write: Operation not permitted
```
See also https://github.com/Homebrew/legacy-homebrew/pull/43972.

**Comment by carlocab** [comment]
Can't reproduce the build failure locally (also Intel Big Sur).

**Comment by iMichka** [comment]
Can you also remove some of the postinstall code, see #75370.
I was not able to fix the pkg-config detection bug in that PR and sort of gave up, but as it looks like they switched to cmake, I hope that this bug is gone now.

File: Formula/watchman.rb [hunk 5]
```diff
@@ -31,53 +14,43 @@ class Watchman < Formula
...
+                    "-DWATCHMAN_STATE_DIR=#{var}/run/watchman",
+                    *std_cmake_args
+
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+
+    path = Pathname.new(File.join(prefix, HOMEBREW_PREFIX))
+    bin.install Dir[path/"bin/*"]
+    lib.install Dir[path/"lib/*"]
+    path.rmtree
```
Comment (carlocab) [inline Formula/watchman.rb:45]: Yep, this is super weird. I'll need to remember to report this upstream.

**Comment by carlocab** [comment]
Done. I hope the build doesn't fail again, as it seems to do so randomly.

**Review by nandahkrishna — APPROVED** [review]

File: Formula/watchman.rb (hunk 5 above)
Comment (carlocab) [reply Formula/watchman.rb:46]: Reported: https://github.com/facebook/watchman/issues/913

**Pull request closed without merging**